openai-core/src/commonMain/kotlin/com.aallam.openai.api/assistant/AssistantTool.kt

```diff
@@ -19,10 +19,18 @@ public sealed interface AssistantTool {
     @SerialName("code_interpreter")
     public data object CodeInterpreter : AssistantTool
 
-
     /**
      * The type of tool being defined: retrieval
      */
+    @Deprecated("For beta assistant-v1 API only")
+    @BetaOpenAI
+    @Serializable
+    @SerialName("retrieval")
+    public data object RetrievalTool : AssistantTool
```

Comment:
Since the whole feature was marked Beta, we can be more aggressive with the API, and simply removed any deprecated APIs related to it